HeavyIonsAnalysis/EventAnalysis/src/HiEvtAnalyzer.cc

```diff
@@ -141,6 +146,7 @@ HiEvtAnalyzer::HiEvtAnalyzer(const edm::ParameterSet& iConfig) :
   VertexTag_(consumes<std::vector<reco::Vertex>>(iConfig.getParameter<edm::InputTag>("Vertex"))),
   puInfoToken_(consumes<std::vector<PileupSummaryInfo>>(edm::InputTag("addPileupInfo"))),
   genInfoToken_(consumes<GenEventInfoProduct>(edm::InputTag("generator"))),
+  generatorlheToken_(consumes<LHEEventProduct>(edm::InputTag("externalLHEProducer",""))),
```

Comment:
This line REQUIRES that an LHEEventProduct be present in the event, or else it errors out. You'll need to protect this line with a boolean.